### PR TITLE
[Buildkite] Update hooks and scripts

### PR DESCRIPTION
**Pull request by mrodm**
## Proposed commit message

This PR updates hooks and scripts used by Buildkite

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- ~[ ] I have verified that all data streams collect metrics or logs.~
- ~[ ] I have added an entry to my package's `changelog.yml` file.~
- ~[ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).~

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: .buildkite/scripts/find_oldest_supported_version.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def handle_or(kibana_version_condition: str):
     conditions = kibana_version_condition.split("||")
     result = ""
     for cond in conditions:
-        candidate = find_oldest_supported_version(cond)
+        candidate = find_oldest_supported_version(cond.strip())
```
Comment (mrodm) [inline .buildkite/scripts/find_oldest_supported_version.py:90]: bug found after declaring first the variable here: https://github.com/elastic/integrations/pull/8482/files#diff-68c0021e761ebe6b8a42d32dfdf3dda2200f0f2fee50106c85c9aea530ccf3c3R355-R356

File: .buildkite/scripts/common.sh [hunk 2]
```diff
@@ -202,16 +205,17 @@ google_cloud_auth_safe_logs() {
 }
 
 google_cloud_logout_active_account() {
-  local active_account=$(gcloud auth list --filter=status:ACTIVE --format="value(account)" 2>/dev/null)
-    if [[ -n "$active_account" && -n "${GOOGLE_APPLICATION_CREDENTIALS+x}" ]]; then
+  local active_account
+  active_account=$(gcloud auth list --filter=status:ACTIVE --format="value(account)" 2>/dev/null || true)
```
Comment (mrodm) [inline .buildkite/scripts/common.sh:209]: Added `|| true` to ensure that this command does not fail if `gcloud` fails or does not exist

**Review by jsoriano — APPROVED** [review]
:+1:

**Review by sharbuz — APPROVED** [review]
:+1:

**Pull request merged**